rest.c

```diff
@@ -976,31 +976,34 @@ restconf_is_list_key_leaf_update (sch_node *api_subtree, GNode *child, GNode *tn
     }
     if (list_parent)
     {
-        char *key = sch_list_key (list_parent);
+        GList *keys = sch_list_keys (list_parent);
         char *list_name = sch_name (list_parent);
         if (g_strcmp0 (list_name, APTERYX_NAME (tnode)) == 0)
         {
             tnode = tnode->children;
         }
         for (GNode *node = g_node_first_sibling (tnode); node; node = g_node_next_sibling (node))
         {
-            if (g_strcmp0 (key, APTERYX_NAME (node)) == 0 && node->children)
+            for (GList *key = keys; key != NULL; key = key->next)
```

Comment:
Not sure I understand what's going on here with this inner for loop. Maybe we could discuss this?